boards/waveshare/rp2040_zero/doc/index.rst

```diff
@@ -0,0 +1,150 @@
+     :align: center
+     :alt: RP2040-Zero
+
+
+.. figure:: img/RP2040-Zero-M-1.jpg
+     :align: center
+     :alt: RP2040-Zero with pre-soldered pin header
+
+     RP2040-Zero (above) and RP2040-Zero with pre-soldered pin header (below)
+     (Images courtesy of Waveshare)
+
+Supported Features
+==================
+
+The RP2040-Zero board configuration supports the following hardware features:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Peripheral
+     - Kconfig option
+     - Devicetree compatible
+   * - NVIC
+     - N/A
+     - :dtcompatible:`arm,v6m-nvic`
+   * - UART
+     - :kconfig:option:`CONFIG_SERIAL`
+     - :dtcompatible:`raspberrypi,pico-uart`
+   * - GPIO
+     - :kconfig:option:`CONFIG_GPIO`
+     - :dtcompatible:`raspberrypi,pico-gpio`
+   * - ADC
+     - :kconfig:option:`CONFIG_ADC`
+     - :dtcompatible:`raspberrypi,pico-adc`
+   * - I2C
+     - :kconfig:option:`CONFIG_I2C`
+     - :dtcompatible:`snps,designware-i2c`
+   * - SPI
+     - :kconfig:option:`CONFIG_SPI`
+     - :dtcompatible:`raspberrypi,pico-spi`
+   * - USB Device
+     - :kconfig:option:`CONFIG_USB_DEVICE_STACK`
+     - :dtcompatible:`raspberrypi,pico-usbd`
+   * - HWINFO
+     - :kconfig:option:`CONFIG_HWINFO`
+     - N/A
+   * - Watchdog Timer (WDT)
+     - :kconfig:option:`CONFIG_WATCHDOG`
+     - :dtcompatible:`raspberrypi,pico-watchdog`
+   * - PWM
+     - :kconfig:option:`CONFIG_PWM`
+     - :dtcompatible:`raspberrypi,pico-pwm`
+   * - Flash
+     - :kconfig:option:`CONFIG_FLASH`
+     - :dtcompatible:`raspberrypi,pico-flash`
+   * - Clock controller
+     - :kconfig:option:`CONFIG_CLOCK_CONTROL`
+     - :dtcompatible:`raspberrypi,pico-clock-controller`
+   * - UART (PIO)
+     - :kconfig:option:`CONFIG_SERIAL`
+     - :dtcompatible:`raspberrypi,pico-uart-pio`
+   * - SPI (PIO)
+     - :kconfig:option:`CONFIG_SPI`
+     - :dtcompatible:`raspberrypi,pico-spi-pio`
+
+Pin Mapping
+===========
+
+The peripherals of the RP2040 SoC can be routed to various pins on the board. The configuration of these routes can be modified through DTS. Please refer to the datasheet to see the possible routings for each peripheral.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+.. rst-class:: rst-columns
+
+- UART0_TX : P0
+- UART0_RX : P1
+- I2C0_SDA : P4
+- I2C0_SCL : P5
+- I2C1_SDA : P6
+- I2C1_SCL : P7
+- SPI0_RX : P16
+- SPI0_CSN : P17
+- SPI0_SCK : P18
+- SPI0_TX : P19
+- ADC_CH0 : P26
+- ADC_CH1 : P27
+- ADC_CH2 : P28
+- ADC_CH3 : P29
+
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+
+Using UF2
+---------
+
+Since it doesn’t expose the SWD pins, you must flash the RP2040-Zero with an UF2 file. By default, building an app for this board will generate a build/zephyr/zephyr.uf2 file. If the RP2040-Zero is powered on with the BOOTSEL button pressed, it will appear on the host as a mass storage device. The UF2 file should be drag-and-dropped to the device, which will flash the RP2040-Zero.
```

Comment:
`west flash` should also work

Reply:
Can it work without the SWD pins exposed? For [similarly shaped](https://docs.zephyrproject.org/latest/boards/seeed/xiao_rp2040/doc/index.html) rp2040 boards that don't expose the debugger pins the Zephyr documentation only mention UF2 flashing.

Reply:
When you have a uf2 runner as the default, west flash will copy the uf2 file over, it has no need for SWD pins, they would only be used with runners that flash using that.

Reply:
Thanks for the explanation. I rewrote the section and now it mentions `west flash` and the manual drag-and-drop.